source/1_engine/rex_engine/src/filesystem/path.cpp

```diff
@@ -0,0 +1,576 @@
+
+      if (path[1] != ':')
+      {
+        return false;
+      }
+
+      if (path[2] != '/' && path[3] != '\\')
+      {
+        return false;
+      }
+
+      return true;
+    }
+    // Returns if the given path is a relative path
+    bool is_relative(rsl::string_view path)
+    {
+      return !is_absolute(path);
+    }
+    // Returns true if the given path points to a file
+    bool is_file(rsl::string_view path)
+    {
+      DWORD attribs = GetFileAttributesA(path.data());
+
+      if (attribs == INVALID_FILE_ATTRIBUTES)
+      {
+        return false;
+      }
+
+      if (attribs & FILE_ATTRIBUTE_DIRECTORY)
+      {
+        return false;
+      }
+
+      return true;
+    }
+    // Returns true if the given path points to a directory
+    bool is_dir(rsl::string_view path)
+    {
+      DWORD attribs = GetFileAttributesA(path.data());
+
+      if (attribs == INVALID_FILE_ATTRIBUTES)
+      {
+        return false;
+      }
+
+      if (attribs & FILE_ATTRIBUTE_DIRECTORY)
+      {
+        return true;
+      }
+
+      return false;
+    }
+    // Returns true if the given path points to a junction
+    bool is_junction(rsl::string_view path)
+    {
+      internal::ReparseTag tag = internal::get_reparse_tag(path);
+      return tag == internal::ReparseTag::Junction;
+    }
+    // Returns true if the given path points to a symlink
+    bool is_link(rsl::string_view path)
+    {
+      internal::ReparseTag tag = internal::get_reparse_tag(path);
+      return tag == internal::ReparseTag::Symlink;
+    }
+    // Returns true if 2 paths point to the same file
+    bool same_file(rsl::string_view path1, rsl::string_view path2)
+    {
+      // simply convert the files into their actual files on disk
+      // then do a string wise comparison
+      rsl::string real_path1 = real_path(path1);
+      rsl::string real_path2 = real_path(path2);
+      return real_path1 == real_path2;
+    }
+    // Returns true if 2 paths point to the same directory
+    bool same_dir(rsl::string_view path1, rsl::string_view path2)
+    {
+      // simply convert the dir into their actual files on disk
+      // then do a string wise comparison
+      rsl::string real_path1 = real_path(path1);
+      rsl::string real_path2 = real_path(path2);
+      return real_path1 == real_path2;
+    }
+    // Splits the path into a head and a tail
+    // the tail is the last pathname component
+    // the head is everything leading up to that
+    SplitResult split(rsl::string_view path)
+    {
+      SplitResult res{};
+
+      // get the last slash position
+      card32 pos = path.find_last_of("/\\");
+
+      // fill in the values
+      res.head = path.substr(0, pos);
+      res.tail = path.substr(pos + 1);
+
+      // return the result
+      return res;
+    }
+    // Splits the path into a head and a tail
+    // the head is either the mount point or an empty string
+    // the tail is everything else
+    SplitResult split_drive(rsl::string_view path)
```

Comment:
There is no drive on Linux and I don't think there are drives on console ( although I am not sure about that ) 
So maybe this functionality does not belong here or should be implemented per platform